types/react-beautiful-dnd/index.d.ts

```diff
@@ -134,18 +136,18 @@ interface Draggable_MapProps {
     shouldAnimateDisplacement?: boolean;
     // only provided when dragging
     // can be null if not over a droppable
-    direction?: Direction;
+    direction: Maybe<Direction>;
     isDropAnimating?: boolean;
     offset?: Position;
-    dimension?: DraggableDimension;
+    dimension: Maybe<DraggableDimension>;
 }
 
 interface Draggable_OwnProps {
     draggableId: DraggableId;
     children: (
         provided: DraggableProvided,
         snapshot: DraggableStateSnapshot
-    ) => React.ReactElement<any>;
+    ) => Maybe<React.ReactNode>;
     index: number;
     isDragDisabled?: boolean;
     disableInteractiveElementBlocking?: boolean;
```

Comment:
@bradleyayers Can you clarify how does it come that in flow these cases were working fine? For example, props isDragDisabled and disableInteractiveElementBlocking from Draggable_OwnProps. This relates to all the other places where I have left `?:`.

Reply:
https://flow.org/en/docs/types/objects/#toc-optional-object-type-properties explains the optional property semantics.

> In addition to their set value type, these optional properties can either be void or omitted altogether. However, they cannot be null.

It looks like `foo?: Foo` in Flow is equivalent to TypeScript, though I'm not sure how "can be void" translates to TypeScript, but I don't think it's significant in our situation.

Here's my understanding of Flow to TypeScript translation:

<table>
<tr><th>Flow</th><th>TypeScript</th></tr>
<tr><td><code>{ foo: Foo }</code></td><td><code>{ foo: Foo }</code></tr>
<tr><td><code>{ foo?: Foo }</code></td><td><code>{ foo?: Foo }</code></tr>
<tr><td><code>{ foo: ?Foo }</code></td><td><code>{ foo: Foo | undefined | null }</code></tr>
<tr><td><code>{ foo?: ?Foo }</code></td><td><code>{ foo?: Foo | undefined | null }</code></tr>
</table>